libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/builder/NavRouteOptionsBuilder.kt

```diff
@@ -0,0 +1,369 @@
+package com.mapbox.navigation.core.routeoptions.builder
+
+import android.content.Context
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.Bearing
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.extensions.applyDefaultNavigationOptions
+import com.mapbox.navigation.base.extensions.applyLanguageAndVoiceUnitOptions
+
+@ExperimentalPreviewMapboxNavigationAPI
+interface NoWaypointsOptionsBuilder {
+    fun fromStartLocation(
+        point: Point,
+        bearing: Double? = null,
+        zLevel: Int?
+    ): WaypointsInProgressBuilder
```

Comment:
```suggestion
    fun fromStartLocation(
      navigationLocation: NavigationLocation,
    ): WaypointsInProgressBuilder
    
    class NavigationLocation private constrictor(
        val point: Point,
        val bearing: Double? = null,
        val zLevel: Int?
    ){
      Builder...
    }
```

This is more flexible and covers java devs needs. Suppose we must use wrappers for all methods